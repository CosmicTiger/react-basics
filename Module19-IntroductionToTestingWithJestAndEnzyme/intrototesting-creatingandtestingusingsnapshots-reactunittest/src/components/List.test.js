import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import List from './List';

configure({
    adapter: new Adapter()
});

describe('Testing Component <List />', () => {
    test('Validating nodes', () => {

        const fruits = [
            { name: 'Strawberry', id: 1 },
            { name: 'Apple', id: 2 },
            { name: 'Orange', id: 3 },
            { name: 'Mango', id: 4 },
        ]

        const wrapper = shallow(<List title='Fuits' list={fruits} />);

        // Validate if a node exists
        expect(wrapper.find('h1').exists()).toBe(true);

        // Validate if it has css class
        expect(wrapper.find('h1').hasClass('big')).toBe(true);

        // Validate quantitys of children elements
        expect(wrapper.find('ul').children().length).toBe(4);

        // Validate the text content
        expect(wrapper.find('ul').childAt(2).text()).toBe('Orange');
        // console.log(wrapper.find('ul').childAt(2).html());

        // Validate the type of node / element
        expect(wrapper.find('ul').childAt(2).type()).toBe('li');

        // Validate HTML
        expect(wrapper.find('ul').childAt(2).html()).toBe('<li>Orange</li>');
    });

    test('Validate updates in props', () => {
        const fruits = [
            { name: 'Strawberry', id: 1 },
            { name: 'Apple', id: 2 },
            { name: 'Orange', id: 3 },
            { name: 'Mango', id: 4 },
        ]

        const wrapper = shallow(<List title='Fuits' list={fruits} />);

        // Validate quantity of li elements
        expect(wrapper.find('li').length).toBe(4);

        // Updating list prop
        wrapper.setProps({
            list: [ 
                { name: 'kiwi', id: 5 }
            ]
        });

        // Validate quantity of li elements
        expect(wrapper.find('li').length).toBe(1);
        
        // Updating Title prop
        wrapper.setProps({
            title: 'Super Fruits'
        });

        expect(wrapper.find('.big').text()).toBe('Super Fruits');
    });

    test('Validate that matches with Snapshots', () => {
        
        const fruits = [
            { name: 'Strawberry', id: 1 },
            { name: 'Apple', id: 2 },
            { name: 'Orange', id: 3 },
            { name: 'Mango', id: 4 },
        ]

        const wrapper = shallow(<List title='Fuits' list={fruits} />);
        
        // we must use enzyme-to-json to make prettier the snapshots
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});