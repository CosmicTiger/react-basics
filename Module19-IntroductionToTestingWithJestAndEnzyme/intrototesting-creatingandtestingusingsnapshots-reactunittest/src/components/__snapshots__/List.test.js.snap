// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component <List /> Validate that matches with Snapshots 1`] = `
<section>
  <h1
    className="big"
  >
    Fuits
  </h1>
  <ul>
    <li
      key="1"
    >
      Strawberry
    </li>
    <li
      key="2"
    >
      Apple
    </li>
    <li
      key="3"
    >
      Orange
    </li>
    <li
      key="4"
    >
      Mango
    </li>
  </ul>
</section>
`;
